src/ProgressMeter.jl

```diff
@@ -330,19 +359,44 @@ function updateProgress!(p::ProgressUnknown; showvalues = (), truncate_lines = f
         return
     end
 
-    if t > p.tlast+p.dt
-        dur = durationstring(t-p.tfirst)
-        msg = @sprintf "%s %d \t Time: %s" p.desc p.counter dur
-        move_cursor_up_while_clearing_lines(p.output, p.numprintedvalues)
-        printover(p.output, msg, p.color)
-        printvalues!(p, showvalues; color = valuecolor, truncate = truncate_lines)
-        flush(p.output)
-        # Compensate for any overhead of printing. This can be
-        # especially important if you're running over a slow network
-        # connection.
-        p.tlast = t + 2*(time()-t)
-        p.printed = true
-        return
+    if p.counter - p.prev_update_count >= p.check_iterations
+        t = time()
+        if p.counter > 2
+            p.check_iterations = calc_check_iterations(p, t)
+        end
+        if t > p.tlast+p.dt
+            dur = durationstring(t-p.tfirst)
+            msg = @sprintf "%s %d \t Time: %s" p.desc p.counter dur
+            move_cursor_up_while_clearing_lines(p.output, p.numprintedvalues)
+            printover(p.output, msg, p.color)
+            printvalues!(p, showvalues; color = valuecolor, truncate = truncate_lines)
+            flush(p.output)
+            # Compensate for any overhead of printing. This can be
+            # especially important if you're running over a slow network
+            # connection.
+            p.tlast = t + 2*(time()-t)
+            p.printed = true
+            p.prev_update_count = p.counter
+            return
+        end
+    end
+end
+
+function is_threading(p::AbstractProgress)
+    length(p.threads_used) > 1 && return true
```

Comment:
I'm not sure of it. Will `Threads.nthreads() == 1 && return false` a plausible shortcut?

Reply:
Good idea!